src/isaw.policy/src/isaw/policy/metadata.py

```diff
@@ -71,24 +99,39 @@ def update(self):
                 if tag:
                     self.metatags.append((u'article:tag', tag.decode('utf8')))
 
-    @property
+    @memoizedproperty
     def image_url(self):
         """Return an image url for the context in that order
         - context image field
         - context lead image field
         - portal logo
         """
         context = aq_inner(self.context)
-        obj_url = context.absolute_url()
-        if hasattr(context, 'getField'):
-            field = self.context.getField('image')
-            if not field:
-                field = context.getField(IMAGE_FIELD_NAME)
+        image_view = context.restrictedTraverse('@@images', None)
+        if image_view is not None:
+            scale = image_view.scale(fieldname=self.image_field, scale='social')
+            if scale is not None:
+                return scale.url
+        return
 
-            if field and field.get_size(context) > 0:
-                return u'%s/%s_%s' % (obj_url, field.getName(), 'thumb')
-
-        return u"%s/isaw_logo.png" % self.portal.absolute_url()
+    @property
```

Comment:
Why not `@memoizedproperty` here?

Reply:
This is only ever called once, so there's no reason to add the memoization overhead, I think.